Framework/Algorithms/inc/MantidAlgorithms/CreateUserDefinedBackground.h

```diff
@@ -0,0 +1,87 @@
+#ifndef MANTID_ALGORITHMS_CREATEUSERDEFINEDBACKGROUND_H_
+#define MANTID_ALGORITHMS_CREATEUSERDEFINEDBACKGROUND_H_
+
+#include "MantidAlgorithms/DllConfig.h"
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/ITableWorkspace_fwd.h"
+#include "MantidHistogramData/Histogram.h"
+namespace Mantid {
+namespace Kernel {
+class Interpolation;
+}
+namespace Algorithms {
+
+/** CreateUserDefinedBackground : Given an input workspace containing data with
+  a background and a table of user-selected points defining the background,
+  creates a new workspace containing background data that can be subtracted
+  from the original data.
+
+  Copyright &copy; 2016 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+  National Laboratory & European Spallation Source
+
+  This file is part of Mantid.
+
+  Mantid is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 3 of the License, or
+  (at your option) any later version.
+
+  Mantid is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  File change history is stored at: <https://github.com/mantidproject/mantid>
+  Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class MANTID_ALGORITHMS_DLL CreateUserDefinedBackground final
```

Comment:
Would you please remove `final` from the class definition and virtual function declaration? To avoid overuse of the `final`, we recently removed these from class_maker.py.